packages/shopify-app-express/src/middlewares/validate-authenticated-session.ts

```diff
@@ -18,17 +18,38 @@ export function validateAuthenticatedSession({
     return async (req: Request, res: Response, next: NextFunction) => {
       config.logger.info('Running validateAuthenticatedSession');
 
-      const sessionId = await api.session.getCurrentId({
-        isOnline: config.useOnlineTokens,
-        rawRequest: req,
-        rawResponse: res,
-      });
+      let sessionId: string | undefined;
+      try {
+        sessionId = await api.session.getCurrentId({
+          isOnline: config.useOnlineTokens,
+          rawRequest: req,
+          rawResponse: res,
+        });
+      } catch (error) {
+        await config.logger.error(
+          `Error when loading session from storage: ${error}`,
+        );
 
-      const session = await config.sessionStorage.loadSession(
-        sessionId as string,
-      );
+        await handleSessionError(req, res, error);
+        return undefined;
+      }
 
-      let shop = req.query.shop;
+      let session: Session | undefined;
+      if (sessionId) {
+        try {
+          session = await config.sessionStorage.loadSession(sessionId);
+        } catch (error) {
```

Comment:
Here we _have_ to catch any errors because the session storage might be user-provided, so we can't tell what kind of errors we might get, and we don't want those to crash the app.